src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -134,6 +134,8 @@ public void HandleRequestHeadersTimeout()
 
         public void HandleReadDataRateTimeout()
         {
+            Debug.Assert(MinRequestBodyDataRate != null);
```

Comment:
If we don't end up making these non-nullable, we should handle the case where it becomes null between the timeout triggering and the handler actually running. I agree with @davidfowl we should make this non-nullable and just remove the feature for HTTP/2 until if/when we implement it.

Reply:
I'm confused what you and David are asking for. It is implemented today. I can be set to null today:

https://github.com/dotnet/aspnetcore/blob/f98959f630f916d74ccdb705bd02ab34b86fcf85/src/Servers/Kestrel/Core/src/Features/IHttpMinRequestBodyDataRateFeature.cs#L17

Reply:
gRPC has used it since .NET Core 3

https://github.com/grpc/grpc-dotnet/blob/ef77760676ce19c279d5480bca68af3ea4d60e96/src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs#L109-L113

Reply:
I completely forgot about that! We have to keep it nullable then. As unlikely as it is, we should just do the guard against the case where it becomes null between the timeout triggering and the handler actually running.